server.rb

```diff
@@ -0,0 +1,189 @@
+require "sinatra/base"
+require "sqlite3"
+require "nokogiri"
+require 'active_support'
+require 'json'
+require 'securerandom'
+
+class Server < Sinatra::Base
+
+  # {
+  #  to: <UUID>
+  #  from: <UUID>
+  #  message: <String>
+  # }
+
+  post '/messages/:format?' do
+    case params['format']
+    when 'json' then
+      message = JSONMessage.new(request.body.read)
+      if message.valid?
+        id = message.save
+
+        message.send_message
+        [200, {message: {id: id}}.to_json]
+      else
+        [400]
+      end
+    when 'xml'
+      message = XMLMessage.new(request.body.read)
+      if message.valid?
+        id = message.save
+        message.send_message
+        builder = Nokogiri::XML::Builder.new do |xml|
+          xml.root do
+            xml.message do
+              xml.id id
+            end
+          end
+        end
+        [200, builder.to_xml]
+      else
+        [400]
+      end
+    else
+      [400]
+    end
+  end
+
+  get '/messages/:id/status/:format?' do
+    status = db.execute("SELECT status from messages WHERE id = '#{params['id']}' LIMIT 1")
+    return [400] if status.empty?
+    case params['format']
+    when 'json'
+      [200, {message: { status: status.flatten.first } }.to_json ]
+    when 'xml'
+      builder = Nokogiri::XML::Builder.new do |xml|
+        xml.root do
+          xml.message do
+            xml.status status.flatten.first
+          end
+        end
+      end
+      [200, builder.to_xml]
+    else
+      [400]
+    end
+  end
+
+  def db
+    SQLite3::Database.new "test.db"
+  end
+
+  class JSONMessage
```

Comment:
I would recommend moving this to a separate file.
For several reasons:
1. Maintainability - it's easier to find where classes are when they have a matching file name.
2. Clearer dependencies between external libs are our own classes. Right now all external libs are required for all the classes. Thus creates really strong coupling.